spark/src/main/spark-3.x/org/apache/comet/shims/ShimCometScanExec.scala

```diff
@@ -69,6 +69,8 @@ trait ShimCometScanExec {
       readSchema: StructType,
       options: ParquetOptions): FileScanRDD =
     classOf[FileScanRDD].getDeclaredConstructors
+      // workaround for aws spark 3.4 implementation
+      .filter(c => List(3,5,6).contains(c.getParameterCount()) )
```

Comment:
Not sure why style checker doesn't find this.

```suggestion
      .filter(c => List(3, 5, 6).contains(c.getParameterCount()) )
```

Reply:
I'm not sure that we should be adding anything specific to closed-source Spark versions without first considering the implications for the project.

I filed https://github.com/apache/datafusion-comet/issues/414 so that we can start that discussion.

This change has no unit test, and I assume that it would be difficult to add a unit test without adding a dependency on the AWS Spark jars, which are maybe not even available except for when we are building on the AWS platform.

Reply:
Yea, I agreed.

I think we shouldn't claim after this PR that Comet can support AWS Spark. The patch itself looks more like a fix to prevent the code uses incorrect constructor which take different parameters.

For the perspective, it looks reasonable to me. But not to say that this goes to support AWS Spark.

Reply:
I am fine with the current workaround, but we can potentially move this function to `spark-3.2`,  `spark-3.3`, and  `spark-3.4` then remove the reflection. Then type check should work...

Reply:
That is also okay.